drivers/clock_control/clock_stm32_ll_u3.c

```diff
@@ -0,0 +1,603 @@
+			return CLOCK_CONTROL_STATUS_OFF;
+		}
+	} else {
+		/* Domain clock sources */
+		if (enabled_clock(pclken->bus) == 0) {
+			return CLOCK_CONTROL_STATUS_ON;
+		} else {
+			return CLOCK_CONTROL_STATUS_OFF;
+		}
+	}
+}
+
+static DEVICE_API(clock_control, stm32_clock_control_api) = {
+	.on = stm32_clock_control_on,
+	.off = stm32_clock_control_off,
+	.get_rate = stm32_clock_control_get_subsys_rate,
+	.get_status = stm32_clock_control_get_status,
+	.configure = stm32_clock_control_configure,
+};
+
+static void set_regu_voltage(uint32_t hclk_freq)
+{
+	if (hclk_freq < MHZ(48)) {
+		LL_PWR_SetRegulVoltageScaling(LL_PWR_REGU_VOLTAGE_SCALE2);
+	} else {
+		LL_PWR_SetRegulVoltageScaling(LL_PWR_REGU_VOLTAGE_SCALE1);
+	}
+}
+
+static void set_up_fixed_clock_sources(void)
+{
+
+	if (IS_ENABLED(STM32_HSE_ENABLED)) {
+		/* Check if need to enable HSE bypass feature or not */
+		if (IS_ENABLED(STM32_HSE_BYPASS)) {
+			LL_RCC_HSE_EnableBypass();
+		} else {
+			LL_RCC_HSE_DisableBypass();
+		}
+
+		/* Enable HSE */
+		LL_RCC_HSE_Enable();
+		while (LL_RCC_HSE_IsReady() != 1) {
+		/* Wait for HSE ready */
+		}
+	}
+
+	if (IS_ENABLED(STM32_HSI_ENABLED)) {
+		/* Enable HSI if not enabled */
+		if (LL_RCC_HSI_IsReady() != 1) {
+			/* Enable HSI */
+			LL_RCC_HSI_Enable();
+			while (LL_RCC_HSI_IsReady() != 1) {
+			/* Wait for HSI ready */
+			}
+		}
+	}
+
+	if (IS_ENABLED(STM32_LSE_ENABLED)) {
+		/* Enable the power interface clock */
+
+		LL_AHB1_GRP2_EnableClock(LL_AHB1_GRP2_PERIPH_PWR);
+
+		if (!LL_PWR_IsEnabledBkUpAccess()) {
+			/* Enable write access to Backup domain */
+			LL_PWR_EnableBkUpAccess();
+			while (!LL_PWR_IsEnabledBkUpAccess()) {
+				/* Wait for Backup domain access */
+			}
+		}
+
+		/* Configure driving capability */
+		LL_RCC_LSE_SetDriveCapability(STM32_LSE_DRIVING << RCC_BDCR_LSEDRV_Pos);
+
+		if (IS_ENABLED(STM32_LSE_BYPASS)) {
+			/* Configure LSE bypass */
+			LL_RCC_LSE_EnableBypass();
+		}
+
+		/* Enable LSE Oscillator */
+		LL_RCC_LSE_Enable();
+		/* Wait for LSE ready */
+		while (!LL_RCC_LSE_IsReady()) {
+		}
+
+		/* Enable LSESYS additionally */
+		LL_RCC_LSE_EnablePropagation();
+		/* Wait till LSESYS is ready */
+		while (!LL_RCC_LSE_IsPropagationReady()) {
+		}
+
+		LL_PWR_DisableBkUpAccess();
+	}
+
+	if (IS_ENABLED(STM32_MSIS_ENABLED)) {
+		/* Set MSIS Range */
+		LL_FLASH_SetLatency(LL_FLASH_LATENCY_2);
+		while(LL_FLASH_GetLatency()!= LL_FLASH_LATENCY_2)
+		{
+		}
+		if(STM32_MSIS_RANGE >= 0 && STM32_MSIS_RANGE <= 3)
+		{
+			LL_PWR_SetRegulVoltageScaling(LL_PWR_REGU_VOLTAGE_SCALE1);
+		}
+		else if(STM32_MSIS_RANGE >= 4 && STM32_MSIS_RANGE <= 7)
+		{
+			LL_PWR_SetRegulVoltageScaling(LL_PWR_REGU_VOLTAGE_SCALE2);
+		}
+
+		/* Enable Epod Booster */
+		LL_AHB1_GRP2_EnableClock(LL_AHB1_GRP2_PERIPH_PWR);
+		LL_RCC_SetEPODBoosterClkSource(LL_RCC_EPODBOOSTCLKSRCE_MSIS);
+		LL_RCC_SetEPODBoosterClkPrescaler(LL_RCC_EPODBOOSTCLKPRESCAL_1);
+		LL_PWR_EnableEPODBooster();
+		while(LL_PWR_IsActiveFlag_BOOST() != 1)
+		{
+		}
```

Comment:
```patch
-		while(LL_PWR_IsActiveFlag_BOOST() != 1)
+		while (LL_PWR_IsActiveFlag_BOOST() != 1U) {
-		{
```

Nitpicking: I would suggest: `while (LL_PWR_IsActiveFlag_BOOST() == 0U)`, for all while loop expecting to test a 0 or non-zero value.